design-proposals/new-proposals-framework.md

```diff
@@ -0,0 +1,74 @@
+# Overview
+
+This proposal outlines a new process for managing KubeVirt Enhancement Proposals (VEPs), emphasizing centralized prioritization and enhanced SIG involvement and collaboration. The proposed changes aim to focus the community's efforts on prioritized pull requests, increase the review bandwidth, and ensure clear visibility of feature progress and associated issues.
+
+## Motivation
+
+KubeVirt’s current approach to proposals is unstructured and has outgrown the community repo, making it difficult to adequately manage both the repo and the design proposals. The current process does not provide a way to prioritize development and review efforts during the development cycle, nor is there a way to derive a roadmap for a specific release. The process also lacks any formal SIG involvement. Being in the community repo also makes it difficult to grow the reviewer and approver list as there are conflicting purposes.
+
+
+With this proposal, we aim to:
+
+- Optimize contributors' and reviewers' bandwidth and experience.
+- Strengthen SIG ownership and accountability.
+- Improve the alignment of enhancements with the project's goals and architecture.
+- Separate the enhancement process from community matters.
+
+# Goals
+
+1. Introduce a new repository, `kubevirt/enhancements`, to house all VEPs.
+2. Prioritize VEPs at the start of each development cycle to streamline review and focus efforts.
+3. Establish SIG-specific review processes with a dedicated reviewer for each VEP.
+4. Create issues to track the progress, maturity, and associated bugs for each accepted VEP.
+5. Establish ownership of the VEP and any additional SIG collaboration.
+
+## Non-Goals
+
+- Removing the central approval mechanism at this stage.
+- Full decentralization of VEP review and approval processes.
+
+## Repos
+
+- **Impacted Repository**: `kubevirt/enhancements` and kubevirt/community
+
+# Design
+
+The process includes the following key components:
+1. **VEP Creation**: VEP authors will initiate proposals via PRs to the `kubevirt/enhancements` repository.
+2. **SIG Review and Collaboration**: Each VEP will have a target SIG, and the SIG will assign a dedicated reviewer to oversee the proposal, collaborate with other SIGs as needed, and provide feedback or veto when necessary.
+3. **Centralized Prioritization**: At the start of each release cycle, a centralized mechanism will prioritize accepted VEPs, focusing community efforts on these.
+4. **Visibility and Tracking**: The Author of an accepted VEPs will open an issue to track their progress, maturity stages (alpha, beta, GA), and list the associated bugs. And user feedback
+5. *Single source of truth**: Each VEP will be the authoritative reference for the associated feature. This aligns with the Kubernetes KEP process. It will ensure that each enhancement
+Includes all the relevant information, including the design and the state.
```

Comment:
Maybe we can emphasize here that whenever the design / api / limitations etc change the VEP should be updated as well with a PR

Reply:
Who is responsible to keep the VEP in sync with the current state of the code?

Reply:
> Who is responsible to keep the VEP in sync with the current state of the code?

As I see it, and this is how it works in Kubernetes as well, the workflow should look something like this:
1. The developer / assignee proposes a VEP, which contains requirements for graduating from alpha to beta.
2. The VEP is approved and merged.
3. The assignee is responsible for implementation.
4. Once the feature is ready to graduate to Beta from the assignee's perspective, a PR will be introduced to update the VEP. The PR will set a target release for Beta graduation and will explain how the requirements for Beta are fulfilled and whether they have slightly changed. The assignee will possibly have to prove it by providing a list of major implementation PRs.
5. Approvers would need to review the VEP, become convinced that the requirements are both good enough for graduation and are truly fulfilled.
6. The VEP will be updated, the feature would graduate.

TLDR: There are two parties here. One is the assignee who's the one having the biggest interest / motivation to get the feature towards graduation. The other is the SIG/project maintainers who focus on graduating only mature enough features.

Therefore, It's the assignee's responsibility to make the updates, and maintainer's responsibility to review and ensure it is properly done.

Reply:
FWIW, it could be possible that assignees change during the lifecycle of the feature (even k8s suffers from this issue). In which case the burden will fall on to SIGs (esp members with admin rights to the repository) to update the issue and keep it in sync.

Reply:
@alaypatel07 I agree.
It's important to emphasize that if the assignee at some point is not willing to promote the VEP anymore, it's the SIG's responsibility to reflect that and keep the VEP in sync, but it's not its responsibility to continue the implementation. If there is enough interest and another assignee is willing to step in, that's great, but if not, the SIG can eventually decide to revert and drop the VEP.